Added feature HTML helper for structure of header element.

CJVIK opened the pull request:
Added an HTML helper for the creation of the header elements. Usage example is shown at end of file.

- [x] Good variable names:
  - [x] camelCase (JS)
  - [x] Purposeful names - easy to understand
- [x] Validating code
- [x] Semantic HTML:
  - [x] Self-sustained containers: article
  - [x] Dependant containers: section
  - [x] Other containers: div

CJVIK committed:
Added HTMLHeader Helper

Created a new file src/Helpers/HTMLHeader.ts to handle the creation of the HTML header and for future use in pages.

CJVIK committed:
Made last changes to usage case

Edited the paragraph to usage example to be inline with Figma design

hjortie commented:
## Notes (and change requests)
- Header tag is never appended to body
- If header tag was to be appended in current state, it would be appended below the script tag; we should probably either move the script tag to head or create the header element as a container in HTML

(also, Linus' name is Lukas 🤪 )

## Checklist
- [x] Good variable names:\
&emsp;\- camelCase (JS)\
&emsp;\- Purposeful names - easy to understand\
- [ ] Validating code
- [x] Semantic HTML:\
&emsp;\- Self-sustained containers: article\
&emsp;\- Dependant containers: section\
&emsp;\- Other containers: div

hjortie reviewed (changes requested):
## Notes (and change requests)
- Header tag is never appended to body
- If header tag was to be appended in current state, it would be appended below the script tag; we should probably either move the script tag to head or create the header element as a container in HTML

(also, Linus' name is Lukas 🤪 )

## Checklist
- [x] Good variable names:\
&emsp;\- camelCase (JS)\
&emsp;\- Purposeful names - easy to understand\
- [x] Validating code
- [x] Semantic HTML:\
&emsp;\- Self-sustained containers: article\
&emsp;\- Dependant containers: section\
&emsp;\- Other containers: div

Ymer-MI commented on src/Helpers/HTMLHeader.ts:
```diff
@@ -0,0 +1,63 @@
+interface SectionTitleContent {
+  h1: string;
```
Semicolon in Object, should be comma sign when listing properties and their data type. Multiple occurrences.

Ymer-MI replied:
Use HTML<tag>Element['innerHTML']  (if existing or the default HTMLElement['innerHTML'] otherwise) if that is what it will be used for. Makes the intention of that parameter more clear when using the function.

Ymer-MI commented on src/Helpers/HTMLHeader.ts:
```diff
@@ -0,0 +1,63 @@
+
+  return header;
+}
+//Exempel användning
```
Put examples in it's own file.

Ymer-MI reviewed (changes requested):
Overall good code. A few suggestions:

1. Declare the variables in as few places as possible to make it easier to find then (you can use const var1, var2, var3 etc.).
2. Group what you are doing together to make it easier to find where a variable gets a certain property or the append structure when working with HTML.
3. If the interfaces isn't used in more than one place, consider dropping it and maybe write it directly as the parameter data type like this for line 14:

```
export function createHeaderContent(content: { sectionTitle: { h1: HTMLHeadingElement['innerHTML'], p?: HTMLParagraphElement['innerHTML'] }, sectionDiscription: { p1: HTMLParagraphElement['innerHTML'], p2?: HTMLParagraphElement['innerHTML'] } }): HTMLElement {
```

Point 1 & 3 also makes the file smaller and faster to load. None of these points makes the code bad, just maybe harder to debug if necessary.

CJVIK committed:
Made clarification in the HTMLHeader

Changed semi-colon to comma in instances of multiple values inline with convetion.
Added clarification to the HTMLHeader with HTMLElement type.
Removed use case.

CJVIK committed:
Optimized createHeaderContent function

Optimized function createHeaderContent in src/Helpers/HTMLHeader.ts
inline with Ymer's code style, pretty neat.
Updated src/pages/parts/header.ts to reflect the changes, and as export const.

CJVIK committed:
Added structure to the header for welcome pages

Made header a separate component and added a header to the welcome page, this way you can make diffrent headers for diffrent pages.

AnnaMargaretaEmelie reviewed (commented):
Inget att tillägga :)

Ymer-MI commented on src/pages/parts/app.ts:
```diff
@@ -1,5 +1,9 @@
 import { head } from './head';
+import { createHeaderContent } from '../../Helpers/HTMLHeader';
```
Helpers should be with a lower case h to match the rest of the file system. Not a break error but still an error. Need for updating the actual folder name to so the paths match completely.

Ymer-MI reviewed (changes requested):
Good and clean update from the first draft. keep it up.

CJVIK committed:
Corrected folder name and path

Corrected folder name to lowercase, and changed path to match the new folder name.

isabellavelasquez reviewed (approved):
looks good! :)

CJVIK committed:
Corrected file name and path

Changed file name of helper folder - and updated path

Ymer-MI reviewed (approved):
Looking good.

hjortie committed:
Merge branch 'dev' into 2-f-header-structure-html
